pkg/config/lima_config_applier.go

```diff
@@ -188,3 +199,50 @@ func hasUserModeEmulationInstallationScript(limaCfg *limayaml.LimaYAML) (int, bo
 
 	return scriptIdx, hasCrossArchToolInstallationScript
 }
+
+func toggleSoci(limaCfg *limayaml.LimaYAML, enabled bool, sociVersion string, arch string) {
+	idx, hasScript := hasSociInstallationScript(limaCfg)
+	fname := fmt.Sprintf("soci-snapshotter-%s-linux-%s.tar.gz", sociVersion, arch)
+	sociDownloadUrl := fmt.Sprintf("https://github.com/awslabs/soci-snapshotter/releases/download/v%s/%s", sociVersion, fname)
+	if !hasScript && enabled {
+		limaCfg.Provision = append(limaCfg.Provision, limayaml.Provision{
+			Mode: "system",
+			Script: fmt.Sprintf(`%s
+if [ ! -f /usr/local/bin/soci ]; then
+    #download soci
```

Comment:
Might need a `sleep 5` here. https://github.com/lima-vm/lima/issues/287

Reply:
I don't think the script needs to wait for any of the mounts to run correctly, so ```sleep 5``` shouldn't be necessary. It doesn't hurt to add a ```sleep``` just in case though, so I'll add it back.

Reply:
If you are sure your provisioning scripts don't deal with mounts, don't need to add it.